Find a functionally equivalent code:com.android.internal.util.XmlUtils.readThisSetXml:COMMENT
Method Modifier: public      static      final       internal    
Comment:/**
 * Read a HashSet object from an XmlPullParser. The XML data could previously
 * have been generated by writeSetXml(). The XmlPullParser must be positioned
 * <em>after</em> the tag that begins the set.
 *
 * @param parser The XmlPullParser from which to read the set data.
 * @param endTag Name of the tag that will end the set, usually "set".
 * @param name An array of one string, used to return the name attribute
 * of the set's tag.
 *
 * @return HashSet The newly generated set.
 *
 * @throws XmlPullParserException
 * @throws java.io.IOException
 *
 * @see #readSetXml
 */

@@ -1,18 +1,3 @@
 {
-    HashSet set = new HashSet();
-    int eventType = parser.getEventType();
-    do {
-        if (eventType == parser.START_TAG) {
-            Object val = readThisValueXml(parser, name);
-            set.add(val);
-        // System.out.println("Adding to set: " + val);
-        } else if (eventType == parser.END_TAG) {
-            if (parser.getName().equals(endTag)) {
-                return set;
-            }
-            throw new XmlPullParserException("Expected " + endTag + " end tag at: " + parser.getName());
-        }
-        eventType = parser.next();
-    } while (eventType != parser.END_DOCUMENT);
-    throw new XmlPullParserException("Document ended before " + endTag + " end tag");
+    return readThisSetXml(parser, endTag, name, null);
 }

